jax/_src/lax/control_flow.py

```diff
@@ -662,7 +662,7 @@ def cond(pred, true_fun, false_fun, operand):
   return _cond(*args, **kwargs)
 
 def _cond(pred, true_fun: Callable, false_fun: Callable, operand):
-  if len(np.shape(pred)) != 0:
+  if not isinstance(pred, Sequence) and len(np.shape(pred)) != 0:
```

Comment:
I find this fix a little confusing. Wouldn't a better fix be:
```
if isinstance(pred, Sequence) or np.ndim(pred) != 0:
...
```

Reply:
Done